camayoc/tests/qpc/cli/csv_report_parsing.py

```diff
@@ -0,0 +1,135 @@
+    # Next, extract header information into a list of dictonaries (1 dictionary
+    # per header set)
+    header_info = extract_key_value_lines(report_headers,
+                                          header_lines)
+    # Now that we extracted the report information we can use CSV reader to
+    # read the system fingerprints information
+    reader = csv.DictReader(f)
+    if report_type == 'summary':
+        report = normalize_summary_report(header_info, reader)
+    else:
+        report = normalize_detail_report(header_info, reader)
+    return report
+
+
+def normalize_summary_report(header_info, reader):
+    """Normalize report info into a summary report.
+
+    Takes information from report_info dict, and reader and returns a
+    summary report
+    """
+    report_info = header_info[0]
+
+    # Ensure extracted fields match expected
+    expected_keys = [x.lower().replace(' ', '_') for x in
+                     EXPECTED_SUMMARY_REPORT_ID_FIELDS]
+    assert sorted(report_info.keys()) == sorted(expected_keys),\
+        "Extracted Report Fields didn't match expected list"
+
+    report = {
+        'report_id': report_info['report_id'],
+        'report_type': report_info['report_type'],
+        'report_version': report_info['report_version'],
+        'report_platform_id': report_info['report_platform_id'],
+        'system_fingerprints': [row for row in reader],
+    }
+    return report
+
+
+def normalize_detail_report(header_info, reader):
+    """Normalize report info into a detail report.
+
+    Takes information from report_info dict, and reader and returns a
+    detail format report
+    """
+    # The first dictionary grabbed contains the report info
+    report_info = header_info[0]
+    # The second contains the source header info
+    source_info = header_info[1]
+
+    # Ensure extracted fields match expected
+    expected_keys = [x.lower().replace(' ', '_') for x in
+                     EXPECTED_DETAIL_REPORT_ID_FIELDS]
+    assert sorted(report_info.keys()) == sorted(expected_keys),\
+        "Extracted Report Fields didn't match expected list"
+
+    report = {
+        'id': report_info['report_id'],
+        'report_type': report_info['report_type'],
+        'sources': [{
+            'facts': [row for row in reader],
+            'server_id': source_info['server_identifier'],
+            'source_name': source_info['source_name'],
+            'source_type': source_info['source_type'],
+        }],
+    }
+    return report
+
+
+def extract_key_value_lines(input_lines, line_pairs, delim=','):
+    """Extract multiple line pairs into list of dictionaries.
+
+    :param input_lines: a list of csv line strings.
+    :param line_pairs: A list of tuples containg the index pairs for the lines
+        to be used as the key/value pairs of each dictionary
+        (ex: [(0,1), (3,4)] ).
+    :param delim: the delimiter to split the lines with (defaults to ',').
```

Comment:
CSV file can escape the `,` character, this is not used currently but is something to be aware when using `split` or other string method to process CSV informaiton.

PS.: You don't need to change anything here, just have this in mind in case the escaped `,` is used in the future.